Binary files ladspa_sdk/src/.makefile.swp and ladspa_sdk-patched/src/.makefile.swp differ
diff -Nurd ladspa_sdk/src/applyplugin.c ladspa_sdk-patched/src/applyplugin.c
--- ladspa_sdk/src/applyplugin.c	2007-11-06 05:52:56.000000000 -0500
+++ ladspa_sdk-patched/src/applyplugin.c	2009-08-22 23:07:11.000000000 -0400
@@ -6,7 +6,7 @@
 /*****************************************************************************/
 
 #include <dlfcn.h>
-#include <endian.h>
+#include <machine/endian.h>
 #include <errno.h>
 #include <math.h>
 #include <stdlib.h>
diff -Nurd ladspa_sdk/src/makefile ladspa_sdk-patched/src/makefile
--- ladspa_sdk/src/makefile	2007-11-06 05:42:45.000000000 -0500
+++ ladspa_sdk-patched/src/makefile	2009-08-22 23:12:41.000000000 -0400
@@ -4,18 +4,18 @@
 #
 # Change these if you want to install somewhere else.
 
-INSTALL_PLUGINS_DIR	=	/usr/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
-INSTALL_BINARY_DIR	=	/usr/bin/
+INSTALL_PLUGINS_DIR	=	@PREFIX@/lib/ladspa/
+INSTALL_INCLUDE_DIR	=	@PREFIX@/include/
+INSTALL_BINARY_DIR	=	@PREFIX@/bin/
 
 ###############################################################################
 #
 # GENERAL
 #
 
-INCLUDES	=	-I.
-LIBRARIES	=	-ldl -lm
-CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
+INCLUDES	=	-I. -I@PREFIX@/include
+LIBRARIES	=	-L@PREFIX@/lib -ldl -lm
+CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC -fno-common
 CXXFLAGS	=	$(CFLAGS)
 PLUGINS		=	../plugins/amp.so				\
 			../plugins/delay.so				\
@@ -27,6 +27,7 @@
 			../bin/listplugins
 CC		=	cc
 CPP		=	c++
+LD		= $(CPP)
 
 ###############################################################################
 #
@@ -35,29 +36,17 @@
 
 ../plugins/%.so:	plugins/%.c ladspa.h
 	$(CC) $(CFLAGS) -o plugins/$*.o -c plugins/$*.c
-	$(LD) -o ../plugins/$*.so plugins/$*.o -shared
+	$(LD) -o ../plugins/$*.so plugins/$*.o -bundle
 
 ../plugins/%.so:	plugins/%.cpp ladspa.h
 	$(CPP) $(CXXFLAGS) -o plugins/$*.o -c plugins/$*.cpp
-	$(CPP) -o ../plugins/$*.so plugins/$*.o -shared
+	$(CPP) -o ../plugins/$*.so plugins/$*.o -bundle
 
 ###############################################################################
 #
 # TARGETS
 #
 
-test:	/tmp/test.wav ../snd/noise.wav always
-	@echo ---------------------------------------------
-	@echo First listen to the white noise input signal:
-	@echo ---------------------------------------------
-	-sndfile-play ../snd/noise.wav
-	@echo -------------------------
-	@echo Compare to plugin output.
-	@echo -------------------------
-	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
-	-sndfile-play /tmp/test.wav
-	@echo Test complete.
-
 install:	targets
 	-mkdirhier $(INSTALL_PLUGINS_DIR)
 	-mkdirhier $(INSTALL_INCLUDE_DIR)
@@ -90,17 +79,17 @@
 #
 
 ../bin/applyplugin:	applyplugin.o load.o default.o
-	$(CC) $(CFLAGS) $(LIBRARIES)					\
+	$(LD) $(CFLAGS) $(LIBRARIES)					\
 		-o ../bin/applyplugin					\
 		applyplugin.o load.o default.o
 
 ../bin/analyseplugin:	analyseplugin.o load.o default.o
-	$(CC) $(CFLAGS) $(LIBRARIES)					\
+	$(LD) $(CFLAGS) $(LIBRARIES)					\
 		-o ../bin/analyseplugin 				\
 		analyseplugin.o load.o default.o
 
 ../bin/listplugins:	listplugins.o search.o
-	$(CC) $(CFLAGS) $(LIBRARIES)					\
+	$(LD) $(CFLAGS) $(LIBRARIES)					\
 		-o ../bin/listplugins	 				\
 		listplugins.o search.o
 
diff -Nurd ladspa_sdk/src/search.c ladspa_sdk-patched/src/search.c
--- ladspa_sdk/src/search.c	2007-11-06 05:48:06.000000000 -0500
+++ ladspa_sdk-patched/src/search.c	2009-08-22 23:07:11.000000000 -0400
@@ -5,12 +5,12 @@
 
 /*****************************************************************************/
 
+#include <sys/types.h>
 #include <dirent.h>
 #include <dlfcn.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/types.h>
 #include <unistd.h>
 
 /*****************************************************************************/
